### ISSUE TITLE
Compose source files

### ISSUE THREAD

**Issue by jhabdas** [xwp/wp-dev-lib#241]
Once #240 is worked out it would be super nice to see some of this stuff rolled into Composer packages, independently versioned and made available on [Packagist](https://packagist.org/).

**Comment by jhabdas** [comment]
@westonruter want me to take a crack at splitting out the readme generator and send up a pull for your review?

**Comment by westonruter** [comment]
Splitting out the readme generator in what way?

**Comment by jhabdas** [comment]
Creating a self-contained composer module. Sorry if the intent of this issue wasn't clear. If you'd prefer no sweat. Not sure what your intent was for bundling and versioning these scripts, if at all.

**Comment by westonruter** [comment]
I think I'd want them to all be contained within the same repo to make it easier to add `dev-lib` to a repo just be adding it as a submodule.

**Comment by jhabdas** [comment]
Agree. I was thinking submodules. What are your thoughts?

**Comment by westonruter** [comment]
I suppose that would work, as long as projects initialize submodules recursively.

**Comment by jhabdas** [comment]
K. I'll look into that first to see if it would cause an issue. Goal here is to keep things painless and easy to maintain and version.

**Comment by jhabdas** [comment]
Looks I was going to try and compose the the readme markdown generator using the following pattern:

```
mkdir -p packages/readme/bin
mv generate-markdown-readme packages/readme/bin
mv generate class-wordpress-readme-parser.php packages/readme
ln -s packages/readme/class-wordpress-readme-parser.php class-wordpress-readme-parser.php
ln -s packages/readme/bin/generate-markdown-readme  generate-markdown-readme
```

After moving the files I was able to create the composer package to test it out and noticed the following already exists on Packagist: https://packagist.org/packages/wpreadme2markdown/wpreadme2markdown

Not sure it'd be worth creating two of a similar thing.

I saw in https://github.com/xwp/wp-dev-lib/issues/113 a contributor was trying to copy patchwork into the lib. Seems that perhaps the real need to to just create a top-level Composer package wrapping the entire dev lib rather than splitting things out. Interested to hear other thoughts.
